src/game/phase/dungeon.cpp

```diff
@@ -14,6 +15,12 @@ void DungeonPhase::s_setup() {
         context.interacts.emplace(dispatcher, other);
     });
 
+    essence_conn = events::dungeon::s_spawn_essence_event.connect([&](float x, float z) {
+        Essence *ess = new Essence();
+        ess->set_position({ x, 0, z });
+        GameState::scene_manager.get_current_scene()->objs.push_back(ess);
+    });
```

Comment:
I disagree, since essence is only a part of the dungeon phase, and should not appear or need to be known anywhere else except for within this phase.